src/lambda/wizards/samDeployWizard.ts

```diff
@@ -413,10 +427,46 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
         const val = picker.verifySinglePickerOutput(choices)
 
         return val?.label && ![loadingBuckets, messages.noBuckets, messages.bucketError].includes(val.label)
-            ? val.label
-            : undefined
+        ? val.label
+        : undefined
     }
 
+    public async promptUserForNewS3Bucket(step: number, selectedRegion: string): Promise<string | undefined>
+    {
+        
+        if (!this.newBucketCalled) {
+            this.additionalSteps++
```

Comment:
Is there any danger of this increasing infinitely if a user keeps entering and exiting this step?

Reply:
No, the `newBucketCalled `boolean is changed to true, so it can only happen once.

Reply:
Does this mess up the step numbers if the user selects a new bucket and then goes back?

Reply:
No, it will only increase the total number of steps by a maximum of 1.